### ISSUE TITLE
MapFile: lseek failed: EBADF (Bad file number)

### ISSUE THREAD

**Issue by hgoebl** [mapsforge/mapsforge#916]
I updated to v0.7.0 and hoped to get rid of this problem, which probably is _not_ a bug of mapsforge:

```
01-01 21:40:59.510 12495-20157/org.myworkouts.debug E/MapFile: lseek failed: EBADF (Bad file number)
   java.io.IOException: lseek failed: EBADF (Bad file number)
       at java.io.RandomAccessFile.seek(RandomAccessFile.java:602)
       at org.mapsforge.map.reader.IndexCache.getIndexEntry(IndexCache.java:91)
       at org.mapsforge.map.reader.MapFile.processBlocks(MapFile.java:496)
       at org.mapsforge.map.reader.MapFile.readMapData(MapFile.java:859)
       at org.mapsforge.map.reader.MapFile.readMapData(MapFile.java:834)
       at org.mapsforge.map.layer.renderer.DatabaseRenderer.executeJob(DatabaseRenderer.java:90)
       at org.mapsforge.map.layer.renderer.MapWorkerPool$MapWorker.run(MapWorkerPool.java:155)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:818)
    Caused by: android.system.ErrnoException: lseek failed: EBADF (Bad file number)
       at libcore.io.Posix.lseek(Native Method)
       at libcore.io.BlockGuardOs.lseek(BlockGuardOs.java:163)
       at java.io.RandomAccessFile.seek(RandomAccessFile.java:600)
       at org.mapsforge.map.reader.IndexCache.getIndexEntry(IndexCache.java:91) 
       at org.mapsforge.map.reader.MapFile.processBlocks(MapFile.java:496) 
       at org.mapsforge.map.reader.MapFile.readMapData(MapFile.java:859) 
...
```

I'm wondering whether it would be possible to catch this exception and handle it somehow.

So the idea would be to catch the exception in the `MapWorker.run` method and mark the tile as "not rendered". Is this possible? Or why is the exception not catched in the thread?

**Comment by devemux86** [comment]
See also #605 where that exception is mentioned too.
Are you sure the map file is valid?

_Mapsforge server had some problems lately and several maps are invalid with small size._

**Comment by ludwigb** [comment]
This is most likely because the file is corrupted.

There are some ways to check that a file is ok before starting the
rendering process, has been discussed on this list before.

Ludwig

**Comment by hgoebl** [comment]
Thank you guys for the help and this awesome piece of software!
In the meanwhile I found the cause of this issue:

My Fragment lifecycle hooks had problematic implementation. In special cases it created a second tile-layer without closing/destroying the old one. I suppose (without really knowing) in this situation the mapfile has been opened by two threads. So the problem was that the Fragement was destroyed, but the same instance has been reused to create it again. The instance has kept some state which collided with the new lifecycle. I don't know why Android calls `onDestroy` but then keeps the instance of the Fragment.

I'm really sorry that I've wasted your time.
